api/src/main/java/jakarta/data/repository/Limit.java

```diff
@@ -113,6 +115,9 @@ public static Limit range(long startAt, long endAt) {
         if (endAt < startAt)
             throw new IllegalArgumentException("startAt: " + startAt + ", endAt: " + endAt);
 
-        return new Limit(endAt - startAt + 1, startAt);
+        if (endAt - startAt >= Integer.MAX_VALUE)
```

Comment:
Does make sense to use a long and have a MAX value on Integer?

Reply:
> Does it make sense to use a long and have a MAX value on an Integer?

The integer maximum is on the range specified between the startAt/endAt, not on the startAt and endAt values themselves.  It does make sense to constrain the maximum range for the reasons discussed earlier in this pull -- anyone needing to return that many results ought to be using pagination, not getting them all at once with a Limit.  Allowing startAt and endAt to be long values might be useful for large data sets, although [JPA](https://jakarta.ee/specifications/persistence/3.0/apidocs/jakarta.persistence/jakarta/persistence/query#setFirstResult(int)) will only allow int.